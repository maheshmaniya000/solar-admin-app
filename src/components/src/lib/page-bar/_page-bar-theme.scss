@use '~@angular/material' as mat;

@mixin theme($theme-or-color-config) {
  $typography-config: mat.get-typography-config($theme-or-color-config);
  $color-config: mat.get-color-config($theme-or-color-config);
  $primary: map-get($color-config, primary);
  $background: map-get($color-config, background);

  sg-page-bar {
    .line {
      background-color: mat.get-color-from-palette($primary, darker);
    }

    .content {
      background-color: mat.get-color-from-palette($background, app-bar);
      @include mat.typography-level($typography-config, caption);
    }
  }
}
