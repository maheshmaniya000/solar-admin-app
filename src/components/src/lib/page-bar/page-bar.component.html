<div class="line"></div>
<div class="content">
  <ng-content></ng-content>
</div>
