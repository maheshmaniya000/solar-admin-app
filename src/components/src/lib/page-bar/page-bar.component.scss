:host {
  display: block;
  width: 100%;
}

.line {
  height: 4px;
}

.content {
  display: none;

  &:not(:empty) {
    display: block;
    height: 28px;
    padding: 0 16px;
  }
}
