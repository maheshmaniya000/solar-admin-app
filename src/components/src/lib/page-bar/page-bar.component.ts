import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'sg-page-bar',
  templateUrl: './page-bar.component.html',
  styleUrls: ['./page-bar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class PageBarComponent {}
