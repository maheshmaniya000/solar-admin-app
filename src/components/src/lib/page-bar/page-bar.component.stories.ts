import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { moduleMetadata, Story } from '@storybook/angular';

import { SgDefaultsModule } from '../defaults/defaults.module';
import { PageBarModule } from './page-bar.module';

const styles = [
  `
  .container {
    width: 100%;
    height: 500px;
    display: flex;
    align-items: center;
  }
`
];
export default {
  title: 'Solargis/Components/Page Bar',
  decorators: [
    moduleMetadata({
      imports: [BrowserAnimationsModule, SgDefaultsModule, MatButtonModule, MatIconModule, PageBarModule]
    })
  ]
};

export const basic: Story = () => ({
  template: `
    <div class="container">
      <sg-page-bar></sg-page-bar>
    </div>
  `,
  styles
});

export const content: Story = () => ({
  template: `
    <div class="container">
      <sg-page-bar>
        <div class="projected-content">
          <span>Toffee apple pie apple pie chocolate cake sweet jujubes halvah</span>
          <button mat-icon-button color="accent">
            <mat-icon>add_circle_outline</mat-icon>
          </button>
        </div>
      </sg-page-bar>
    </div>
  `,
  styles: [
    ...styles,
    `
    .container {
      background-color: white;
    }
    
    button {
      margin-left: 20px;
    }
    
    .projected-content {
      display: flex;
      align-items: center;
      max-height: 100%;
    }
  `
  ]
});
