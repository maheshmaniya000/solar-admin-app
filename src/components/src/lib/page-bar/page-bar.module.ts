import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { PageBarComponent } from './page-bar.component';

@NgModule({
  declarations: [PageBarComponent],
  imports: [CommonModule],
  exports: [PageBarComponent]
})
export class PageBarModule {}
