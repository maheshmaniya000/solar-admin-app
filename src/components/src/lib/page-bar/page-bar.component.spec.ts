import { createHostFactory, Spectator } from '@ngneat/spectator/jest';

import { PageBarComponent } from './page-bar.component';

describe('PageBarComponent', () => {
  describe('component', () => {
    let spectator: Spectator<PageBarComponent>;
    const createHost = createHostFactory({
      component: PageBarComponent
    });

    it('should match snapshot without content', () => {
      spectator = createHost(`<sg-page-bar></sg-page-bar>`);
      expect(spectator.fixture).toMatchSnapshot();
    });

    it('should match snapshot with content', () => {
      spectator = createHost(`<sg-page-bar><div>some <b>content</b>'</div></sg-page-bar>`);
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
