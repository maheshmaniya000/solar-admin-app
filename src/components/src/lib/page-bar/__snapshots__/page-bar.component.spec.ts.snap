// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBarComponent component should match snapshot with content 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-page-bar>
    <div
      class="line"
    />
    <div
      class="content"
    >
      <div>
        some 
        <b>
          content
        </b>
        '
      </div>
    </div>
  </sg-page-bar>
</ng-component>
`;

exports[`PageBarComponent component should match snapshot without content 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-page-bar>
    <div
      class="line"
    />
    <div
      class="content"
    />
  </sg-page-bar>
</ng-component>
`;
